proposals/0000-extract-runtime.md

```diff
@@ -0,0 +1,421 @@
+This is (non exclusively) caused by the fact that the scope of changes between version goes 
+across many API surfaces. Having the (limited) possibility to update the runtime version 
+in a project independently of UI would make it more approachable by reducing the all-or-nothing 
+upgrading pattern.
+
+Additionally, there are quite a few missing pieces in the current ecosystem, that from what I know, 
+are hard to add for different reasons, such as `i18n`, `subtleCrypto` among others. There is also 
+an ongoing effort to increase the compatibility with web environments by bringing the 
+[WebAPIs to the React Native](https://github.com/microsoft/rnx-kit/pull/2504). 
+
+Having a package that can iterate separately, working on adding missing features
+independently of React Native UI layer would make bringing the support faster.
+Currently, React Native tightly couples its feature and UI code, making them non-separable. 
+Given that React Native plans on integrating more closely with WebAPIs (or a subset), this
+coupling makes it hard to work on these APIs as well as test them, slowing down the development speed,
+as we need to create a UI to even be able to test them, or otherwise is very hard to do (as it seems
+to not be a supported development path).
+
+It should be possible to alter or swap some parts of React Native without creating and maintaining
+a fork or a long list of internal patches. A plug-and-play architecture is a sign of good design,
+allows greater modularity, and facilities separation of concerns and decoupling.
+
+### Goals
+
+**The main goal is to make it possible for end-users, by altering build and runtime configuration,
+to create perfectly fit runtime variant they need for the application.**
+
+> __NOTE__: In this context we are talking about headless APIs, not related to UI.
+
+To do this, we should have an extensible layer, that can be built upon and configured, just like LEGO blocks. 
+
+For example, if the project is in need for the i18n API, and they are ready to accept the 
+drawbacks of increased app binary and increased app startup time, it should be easily doable, 
+without forking the React Native. The same can be said about other APIs, such as `crypto`, 
+where they might want to use our provided solution, not use it at all, or use in-house closed 
+source solution.
+
+The same can be said about changing the JS engine. The proposed layer would take care of allowing user
+to use another engine by using the already existing `jsi` interface. JS engine should be treated
+like a building block similar to others that can be changed easily by build time configuration.
+
+Right now, we have something similar already: the C++-only TurboModules, which are kind of tightly coupled 
+with the rest of the React Native and the distinction can be a bit confusing for newcomers.
+
+Such blocks would have access to Event Loop and JS engine, so new globals can be added in as-needed
```

Comment:
What "blocks"? Is it a followup to the LEGO blocks analogy? Let's make it clear what a block is and if possible don't introduce a new concept for that

Reply:
Fixed the wording